classes/dojo.py

```diff
@@ -0,0 +1,133 @@
+        else:
+            room_type = room_type.lower()
+            room_name = room_name.lower()
+        if room_type != "office" and room_type != "living_space":
+            raise ValueError("Invalid room type: Your room type should be 'office' or 'living_space'")
+        existing_rooms = []
+        for room in self.all_rooms:
+            if room.room_name == room_name and room.room_type == room_type:
+                existing_rooms.append({room_name: room_type})
+        if {room_name: room_type} in existing_rooms:
+            print (room_type.title() + " " + room_name.title() + " already exists")
+        else:
+            if room_type == 'office':
+                room = Office(room_name)
+                self.offices.append(room)
+            elif room_type == 'living_space':
+                room = LivingSpace(room_name)
+                self.living_spaces.append(room)
+            self.all_rooms.append(room)
+            print ("An " + room.room_type + " called " + room.room_name.title() + " has been successfully created!")
+
+    def add_person(self, person_type, person_name, person_surname, wants_accommodation="N"):
+        if wants_accommodation:
+            wants_accommodation = wants_accommodation.lower()
+            if wants_accommodation != "y" and wants_accommodation != "n":
+                raise ValueError("Please input Y or N for wants_accommodation")
+        if not isinstance(person_type, str) or not isinstance(person_name, str) or not isinstance(person_surname, str):
+            raise ValueError('Only strings are allowed as input')
+        else:
+            person_type = person_type.lower()
+            person_name = person_name.lower()
+            person_surname = person_surname.lower()
+            if person_type != 'staff' and person_type != 'fellow':
+                raise ValueError("Invalid person_type: Your person type should be either 'fellow' or 'staff'")
+            if person_type == 'staff':
+                person = Staff(person_name, person_surname)
+            elif person_type == 'fellow':
+                person = Fellow(person_name, person_surname, wants_accommodation)
+            existing_persons = []
+            if self.all_persons:
+                for item in self.all_persons:
+                    if item.person_name == person.person_name and item.person_surname == person.person_surname:
+                        existing_persons.append(person)
+            if person in existing_persons:
+                print (person.person_name.title() + " " + person.person_surname.title() + " already exists")
+            else:
+                self.all_persons.append(person)
+                print (person.person_type.title() + " " + person.person_name.title() + " " + person.person_surname.title() + " has been successfully added!")
+
+    def allocate_rooms(self):
+        if self.all_persons:
+            for person in self.all_persons:
+                if person not in self.allocated_persons:
```

Comment:
Fixed in the new commit. I'm now checking for this only once.
The excessive nesting has also been reduced by moving the else statements checking if lists have contents to (near) the top of the function